src/Microsoft.IdentityModel.Tokens/JsonWebKeySet.cs

```diff
@@ -117,73 +116,151 @@ public JsonWebKeySet(string json, JsonSerializerSettings jsonSerializerSettings)
-                            }
-                            catch (CryptographicException ex)
-                            {
-                                throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX10802, webKey.X5c[0]), ex));
-                            }
-                            catch (FormatException fex)
-                            {
-                                throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX10802, webKey.X5c[0]), fex));
-                            }
-                        }
+                        AddX509SecurityKey(signingKeys, webKey);
+                        rsaKeyAdded = true;
                     }
 
                     if (!string.IsNullOrWhiteSpace(webKey.E) && !string.IsNullOrWhiteSpace(webKey.N))
                     {
-                        try
-                        {
-                            SecurityKey key =
-                                 new RsaSecurityKey
-                                 (
-                                    new RSAParameters
-                                    {
-                                        Exponent = Base64UrlEncoder.DecodeBytes(webKey.E),
-                                        Modulus = Base64UrlEncoder.DecodeBytes(webKey.N),
-                                    }
-
-                                );
-                            key.KeyId = webKey.Kid;
-                            keys.Add(key);
-                        }
-                        catch (CryptographicException ex)
-                        {
-                            throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX10801, webKey.E, webKey.N), ex));
-                        }
-                        catch (FormatException ex)
-                        {
-                            throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX10801, webKey.E, webKey.N), ex));
-                        }
+                        AddRsaSecurityKey(signingKeys, webKey);
+                        rsaKeyAdded = true;
+                    }
+
+                    if (!rsaKeyAdded)
+                    {
+                        // an rsa key was not resolved, but that doesn't mean that it's necessarily invalid.
+                        LogHelper.LogInformation(LogHelper.FormatInvariant(LogMessages.IDX10810, webKey.KeyId ?? ""));
+                        signingKeys.Add(webKey);
+                    }
+                }
+                else if (webKey.Kty != null && webKey.Kty.Equals(JsonWebAlgorithmsKeyTypes.EllipticCurve, StringComparison.Ordinal))
+                {
+                    try
+                    {
+                        if (ecdsaAdapter == null)
+                            ecdsaAdapter = new ECDsaAdapter();
```

Comment:
could this be static to the class rather than instance?

Reply:
ECDsa is not supported in our NETSTANDARD1.4 target, when running on platforms other than Windows. We'll thrown PlatformNotSupportedException when users try to initialize the adapter and it would be a breaking change for our netstandard1.4 mac/linux users.

That being said, we can initialize the adapter in a static constructor and swallow the exception (log it, perhaps).

Reply:
addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1143/commits/80717334871e2c108fa390f73b5aec1257b126ff